x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleService.java

```diff
@@ -57,7 +57,7 @@ synchronized void shutDown() {
         this.state.set(WatcherState.STOPPING);
         shutDown = true;
         clearAllocationIds();
-        watcherService.shutDown();
+        watcherService.shutDown(null);
```

Comment:
personal preference: can you not make it nullable but pass an empty listener instead?

Reply:
done in https://github.com/elastic/elasticsearch/pull/43939/commits/39b0336b38555d3a2590c063068cd595ee327210